src/Elastic.Transport/Components/Serialization/TransportSerializerExtensions.cs

```diff
@@ -2,6 +2,12 @@
 // Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
 // See the LICENSE file in the project root for more information
 
+using System.IO;
+using System.Text.Json;
+using System;
+using System.Text;
+using System.Text.Json.Nodes;
+
 namespace Elastic.Transport.Extensions;
```

Comment:
Fixed nullability annotations for the existing `SerializeToBytes()` and `SerializeToString()` methods, added `(object?, Type)` overloads for consistency and made `SerializeToString()` aware of `SystemTextJsonSerializer` for improved efficiency.